File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    var answer = '';
    var result = [];

    let surveyMap = new Map();

    surveyMap.set("R", 0);
    surveyMap.set("T", 0);
    surveyMap.set("C", 0);
    surveyMap.set("F", 0);
    surveyMap.set("J", 0);
    surveyMap.set("M", 0);
    surveyMap.set("A", 0);
    surveyMap.set("N", 0);

    survey.map((value, index) => {
        if(choices[index] == 1) surveyMap.set(value[0], surveyMap.get(value[0]) + 3);       
        else if(choices[index] == 2) surveyMap.set(value[0], surveyMap.get(value[0]) + 2); 
        else if(choices[index] == 3) surveyMap.set(value[0], surveyMap.get(value[0]) + 1);
        else if(choices[index] == 4) {}
        else if(choices[index] == 5) surveyMap.set(value[1], surveyMap.get(value[1]) + 1);
        else if(choices[index] == 6) surveyMap.set(value[1], surveyMap.get(value[1]) + 2);
        else if(choices[index] == 7) surveyMap.set(value[1], surveyMap.get(value[1]) + 3);
    });

    for(var i=0; i<[...surveyMap].length; i+=2){
        //console.log(`now surveyMap[i][0]: ${[...surveyMap][i]}`)
        if([...surveyMap][i][1] > [...surveyMap][i+1][1]) result.push([...surveyMap][i][0]);
        else if([...surveyMap][i][1] < [...surveyMap][i+1][1]) result.push([...surveyMap][i+1][0]);
        else if([...surveyMap][i][1] == [...surveyMap][i+1][1]){
            //console.log("same score!!");
            //console.log([...surveyMap].join().charCodeAt(i*4), [...surveyMap].join().charCodeAt(i*4+4));
            result.push([...surveyMap][i][0]);
        }
    }

    answer = result.join('');
    //console.log(result);
    return answer;
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.swift
import Foundation

func solution(_ survey:[String], _ choices:[Int]) -> String {
    var result = ""
    var charPoint = [Character: Int]()
    let sortedAlphaList: [Character] = ["R", "T", "C", "F", "J", "M", "A", "N"]
    let half = 4
    sortedAlphaList.forEach { charPoint[$0] = 0 }
    for (i, sv) in survey.enumerated() {
        // guard choices[i] != 4 else { continue }
        if choices[i] == 4 { continue }
        for (j, char) in sv.enumerated() {
            let leftRight = choices[i] < half ? 0 : 1
            guard j == leftRight else {
                // charPoint[char] = 0
                continue
            }
            var point = 0
            switch choices[i] {
            case 1, 7:
                point = 3
            case 2, 6:
                point = 2
            case 3, 5:
                point = 1
            default:
                point = 0
            }
            charPoint[char] = (charPoint[char] ?? 0) + point
            break
        }
    }
    for i in stride(from: 0, to: sortedAlphaList.count, by: 2) {
        guard i < sortedAlphaList.count - 1 else { break }
        if let leftPoint = charPoint[sortedAlphaList[i]],
            let rightPoint = charPoint[sortedAlphaList[i + 1]] {
            if leftPoint >= rightPoint {
                result += String(sortedAlphaList[i])
            } 
            if leftPoint < rightPoint {
                result += String(sortedAlphaList[i + 1])
            }
            if result.count == 4 {
                break
            }
        }
    }
    return result
}